projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php

```diff
@@ -430,12 +430,17 @@ public function add_options_menu( $wp_admin = false ) {
 		$this->update_submenus( 'options-general.php', array( 'options-general.php' => 'https://wordpress.com/settings/general/' . $this->domain ) );
 		add_submenu_page( 'options-general.php', esc_attr__( 'Advanced General', 'jetpack' ), __( 'Advanced General', 'jetpack' ), 'manage_options', 'options-general.php', null, 1 );
 
+		add_submenu_page( 'options-general.php', esc_attr__( 'Performance', 'jetpack' ), __( 'Performance', 'jetpack' ), 'manage_options', 'https://wordpress.com/settings/performance/' . $this->domain, null, 2 );
+
 		if ( $wp_admin ) {
 			return;
 		}
 
-		$this->hide_submenu_page( 'options-general.php', 'options-discussion.php' );
-		$this->hide_submenu_page( 'options-general.php', 'options-writing.php' );
+		$submenus_to_update = array(
+			'options-writing.php'    => 'https://wordpress.com/settings/writing/' . $this->domain,
```

Comment:
Small nitpick: what if we move strings like `'https://wordpress.com/settings/writing/' . $this->domain` and `'https://wordpress.com/settings/discussion/' . $this->domain` into properties initialised in the constructor (e.g. $calypso_settings_writing_url)? This way we remove duplicating the URLs in class-jetpack-admin-menu.php:230

Reply:
I think this is one of these cases where I prefer to sacrifice maintainability over readability and consistency. I don't like maintaining duplicated fragments of code, but it's something I'm willing to accept if the resulting code of avoiding it ends up being inconsistent with how the rest of the code behaves and/or forces the developer to jump back-and-forth into multiple source files in order have a fully understanding of the logic.

However, if folks feel strongly against this, I'm happy to refactor the code and get rid of the duplications.